src/hypertable_restrict_info.c

```diff
@@ -220,48 +297,46 @@ hypertable_restrict_info_get(HypertableRestrictInfo *hri, AttrNumber attno)
 	return NULL;
 }
 
+typedef List *(*get_const_expr_values) (Const *c);
+
 static bool
-hypertable_restrict_info_add_op_expr(HypertableRestrictInfo *hri, PlannerInfo *root, OpExpr *clause)
+hypertable_restrict_info_add_expr(HypertableRestrictInfo *hri, PlannerInfo *root, List *expr_args, Oid op_oid, get_const_expr_values const_values, bool use_or)
```

Comment:
I think passing const_values here makes this harder to read than necessary. Can't we change this function to return a const * (or NULL) and have the calling function (hypertable_restrict_info_add_restrict_info) call either get_scalar_op_expr_values or get_op_expr_values as appropriate since it knows which one to call?

Reply:
I see two problems with that approach: 1. there are various checks and transformations that happen in `hypertable_restrict_info_add_expr` before we can actually try to extract the values 2. we will upfront pay the price for extracting the values even if we fail on some checks after